snarkos/main.rs

```diff
@@ -141,20 +127,34 @@ async fn start_server(config: Config) -> anyhow::Result<()> {
         Duration::from_secs(config.p2p.peer_sync_interval.into()),
     )?;
 
-    // Construct the server instance. Note this does not start the server.
+    // Construct the node instance. Note this does not start the network services.
     // This is done early on, so that the local address can be discovered
     // before any other object (miner, RPC) needs to use it.
-    let mut server = Server::new(environment.clone()).await?;
+    let mut node = Node::new(environment.clone()).await?;
+
+    // Construct the consensus instance and set it on the node instance.
+    let consensus = Consensus::new(
+        node.clone(),
+        Arc::new(RwLock::new(storage)),
```

Comment:
every time I encounter this, I'm double-checking whether this is the first time we've used the `storage` object; we should probably re-bind it, wrapping it in an `Arc<RwLock<T>>`, as soon as it's been used to create `memory_pool` to avoid any confusion; this should be left to a follow-up, however, as I'm seeing potential for further related improvements

Reply:
done in https://github.com/AleoHQ/snarkOS/pull/618